### PR TITLE
support bf16

### PR DESCRIPTION
**Pull request by asleepzzz**
we modify driver to rocdl and support bf16,
but we do not yet implement host transform value to bf16

there are still so many unit tests we not modify(mfma bf16 already changed ,so unit tests must changed),
ex:
Conversion/GPUToROCDL/gpu-to-rocdl.mlir
Conversion/MIOpenToGPU/mfma_v2.mlir
Dialect/GPU/ops.mlir
Dialect/MIOpen/ops_2.mlir

### CODE REVIEW

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

**Comment by asleepzzz** [comment]
> Can one of the admins verify this patch?

 this PR is still in progress ,I and Letao finish most part of bf16 task, but there are still some 
verification not complete

**Comment by whchung** [comment]
waiting for one end-to-end (compile + execution) example.

**Comment by asleepzzz** [comment]
run:
./bin/mlir-miopen-driver -p -x2 -t bf16 -ph -c| ./bin/mlir-rocm-runner --shared-libs=./lib/librocm-runtime-wrappers.so,./lib/libmlir_runner_utils.so --entry-point-result=void
but verification not complete yet

**Comment by jerryyin** [comment]
@asleepzzz Could you register a user in our CI site at http://ml-ci.amd.com:21096/? Once I got your user id, I will add you to the admin list such that your PR will trigger a CI run.

**Comment by asleepzzz** [comment]
> @asleepzzz Could you register a user in our CI site at http://ml-ci.amd.com:21096/? Once I got your user id, I will add you to the admin list such that your PR will trigger a CI run.

name:hanchang
account:asleepzzz

**Comment by jerryyin** [comment]
@asleepzzz Thanks, I have added your account id to admin list, now your new commit will trigger CI runs.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: remove `b.getBF16Type()` check.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: remove `b.getBF16Type()` check.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: remove b.getBF16Type() check.

File: mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp
Comment (whchung) [inline mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp]: the naming of the function is confusing. please revise it.

Comment (whchung) [reply mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp]: it seems the function is not used anywhere. please remove it if that's really the case.

**Review by whchung — CHANGES_REQUESTED** [review]
some minor changes are needed.
also fix help fix errors found in CI

**Comment by whchung** [comment]
and please help resolve merge conflicts

**Comment by whchung** [comment]
will hold off merging this PR until #53 gets merged.

**Comment by whchung** [comment]
@asleepzzz / @ltqin please help rebase the PR to `miopen-dialect`, we have migrated to using a newer LLVM.

**Pull request closed without merging**
